async-openai/src/types/chat.rs

```diff
@@ -764,11 +764,34 @@ pub struct ChatChoiceStream {
     /// The index of the choice in the list of choices.
     pub index: u32,
     pub delta: ChatCompletionStreamResponseDelta,
+    #[serde(deserialize_with = "deserialize_finish_reason")]
     pub finish_reason: Option<FinishReason>,
     /// Log probability information for the choice.
     pub logprobs: Option<ChatChoiceLogprobs>,
 }
 
+fn deserialize_finish_reason<'de, D>(deserializer: D) -> Result<Option<FinishReason>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    match s.as_str() {
+        "stop" => Ok(Some(FinishReason::Stop)),
+        "length" => Ok(Some(FinishReason::Length)),
+        "tool_calls" => Ok(Some(FinishReason::ToolCalls)),
+        "content_filter" => Ok(Some(FinishReason::ContentFilter)),
+        "function_call" => Ok(Some(FinishReason::FunctionCall)),
+
+        // whitelist known incompatible values
+        "" => Ok(None),
+
+        _ => Err(serde::de::Error::custom(format!(
+            "unknown finish reason: {}, expected one of: stop, length, tool_calls, content_filter, function_call",
+            s
+        ))),

```

Comment:
I suggest we just whitelist all values for better compatibility.

Reply:
accepting all unknown values as None?